traits.tests.ts

```diff
@@ -0,0 +1,873 @@
+    );
+
+    const expectedTraitReferenceMap = new Map(
+      Object.entries({
+        "test-trait": { token: "trait_reference" },
+      })
+    );
+
+    // Act
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
+
+    // Assert
+    expect(traitReferenceMap).toEqual(expectedTraitReferenceMap);
+  });
+
+  it("correctly builds the trait reference map for a direct trait that is the second parameter", () => {
+    // Arrange
+    const allFunctionsInterfaces = JSON.parse(
+      readFileSync(
+        "./fixtures/direct-trait-2nd-parameter-functions-interfaces.json",
+        "utf-8"
+      )
+    );
+
+    const expectedTraitReferenceMap = new Map(
+      Object.entries({
+        "test-trait": { token: "trait_reference" },
+      })
+    );
+
+    // Act
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
+
+    // Assert
+    expect(traitReferenceMap).toEqual(expectedTraitReferenceMap);
+  });
+
+  it("correctly builds the trait reference map for a direct trait that is the fifth parameter", () => {
+    // Arrange
+    const allFunctionsInterfaces = JSON.parse(
+      readFileSync(
+        "./fixtures/direct-trait-5th-parameter-functions-interfaces.json",
+        "utf-8"
+      )
+    );
+
+    const expectedTraitReferenceMap = new Map(
+      Object.entries({
+        "test-trait": { e: "trait_reference" },
+      })
+    );
+
+    // Act
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
+
+    // Assert
+    expect(traitReferenceMap).toEqual(expectedTraitReferenceMap);
+  });
+
+  it("correctly builds the trait reference map for a tuple nested trait that is the first parameter", () => {
+    // Arrange
+    const allFunctionsInterfaces = JSON.parse(
+      readFileSync(
+        "./fixtures/tuple-trait-1st-parameter-functions-interfaces.json",
+        "utf-8"
+      )
+    );
+
+    const expectedTraitReferenceMap = new Map(
+      Object.entries({
+        "test-trait": {
+          "tuple-param": { tuple: { token: "trait_reference" } },
+        },
+      })
+    );
+
+    // Act
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
+
+    // Assert
+    expect(traitReferenceMap).toEqual(expectedTraitReferenceMap);
+  });
+
+  it("correctly enriches interface with trait reference data for a direct trait that is the first parameter", () => {
+    // Arrange
+    const targetContractId = "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.trait";
+
+    const allFunctionsInterfaces = JSON.parse(
+      readFileSync("./fixtures/direct-trait-functions-interfaces.json", "utf-8")
+    ).filter((f: any) => f.name !== "update-context");
+
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
+
+    const ast = JSON.parse(
+      readFileSync("./fixtures/direct-trait-ast.json", "utf-8")
+    );
+
+    const expected = new Map(
+      Object.entries({
+        [targetContractId]: [
+          {
```

Comment:
For example, here it's very nice that the reader can see the json without searching in files.